libraries/AP_MSP/AP_MSP_Telem_Backend.cpp

```diff
@@ -906,11 +906,10 @@ MSPCommandResult AP_MSP_Telem_Backend::msp_process_out_analog(sbuf_t *dst)
     } analog {
         voltage_dv : (uint8_t)constrain_int16(battery_state.batt_voltage_v * 10, 0, 255),                   // battery voltage V to dV
         mah : (uint16_t)constrain_int32(battery_state.batt_consumed_mah, 0, 0xFFFF),                        // milliamp hours drawn from battery
-        rssi : uint16_t(get_rssi(rssi) ? (uint16_t)constrain_float(rssi,0,1) * 1023 : 0),                   // rssi 0-1 to 0-1023),                           // rssi 0-1 to 0-1023
+        rssi : uint16_t(get_rssi(rssi) ? constrain_float(rssi,0,1) * 1023 : 0),                             // rssi 0-1 to 0-1023)
```

Comment:
```suggestion
        rssi : uint16_t(get_rssi(rssi) ? rssi * 1023 : 0),                             // rssi 0-1 to 0-1023
```

Reply:
redundant? might be, the only type I have doubts about is RSSI_TYPE 3, fport and fport2 should be ok, but get_receiver_rssi() relies on protocol handler to guarantee the 0-1 range
https://github.com/ArduPilot/ardupilot/blob/c425a75ceac2b6291e4af0a1fa2012cc42704e81/libraries/AP_RSSI/AP_RSSI.cpp#L132-L155

Reply:
looks like constrain may be needed